Link features

HubertReX opened the pull request:
## Summary

This PR implements issue #246 - more description can be found inside the ticket.


## Checklist

- [x] I have tested this change locally and it works as expected.
- [x] I have made sure that the code follows the formatting and style guidelines of the project.

## Labels
`type: feature`, `area: gameplay`

DangerousVanilla commented on main.py:
```diff
@@ -90,16 +101,41 @@ def __init__(self):
-        self.round_end_timer = 0.0
-        self.round = 1
+        # config of all game versions and all rounds: rounds_config[game_version][round_no][feature_name] = value
+        self.rounds_config: list[list[dict[str, Any]]] = support.load_data(
```
Spoke with HuberReX and was able to resolve this by building the appropriate round config file with the script.

DangerousVanilla commented:
@HubertReX, thanks again for working with me today. I've been through each version of the game and there are only a couple small things I noticed in version 3:
The player still has a hat but it should be hidden in v3 of the game.
![image](https://github.com/user-attachments/assets/910bc7f0-1b2d-4412-aa43-0892c30a92e0)
![hat in v3](https://github.com/user-attachments/assets/de84b579-bba5-4d93-af29-c28ea1b7dd78)


NPC's still cheer for both the player and the NPC opponent in the cow herding minigame but they should also be hidden in v3.
![image](https://github.com/user-attachments/assets/a4acbad8-5378-4959-802d-f594f0af7631)
![cheering_characters_v3](https://github.com/user-attachments/assets/15075247-390e-4791-8b8d-1c3526e12619)



Once we're up to date with main so I can properly test the scripted sequences as well I'll be able to tell if we have any remaining conflicts with the intended functionality.

DangerousVanilla commented:
@HubertReX, just wanted to give you one more thing I've found so far - it looks like the collision boxes for the hidden bath house signs still persist. If you try to walk through where they are normally you still get stuck on them:

![sign_collision_boxes_persist](https://github.com/user-attachments/assets/25653965-6892-44d3-a99d-1a2b98ae1e29)

HubertReX commented:
@DangerousVanilla 
I've fixed all above-mentioned bugs, and I also added more features - sorry to make it an even bigger PR, but it was all connected, and I didn't want to wait with work till the merge.

New features since the last tests on 20th of December:
- keybindings for debugging (e.g. end of round `r`) are now hidden in options if game version is not DEBUG
- new way of hiding bath signs (both visual and collision objects are removed)
- shop menu improvements (scroll, icons, crop_types_list features)
- last step of tutorial is hidden if playable_outgroup feature is disabled
- implemented crop_types_list feature (#248)
- system-wide translations (see data/translations/en.txt, data/textboxes/en/dialouges.json and tutorial.json) - use `run_game de` to run game using German translations.

DangerousVanilla reviewed (approved):
@HubertReX, looks like this is working as intended, thank you!

I did notice one tiny thing that came up after the implementation of the keybindings hiding in non-debug modes. That is, when the menu items are hidden in the keybinding list the scroll area isn't being recalculated based on the new number of objects so there's a bunch of blank space underneath. 

Since this isn't affecting functionality in any way I'm going to approve and merge this monster as is and this tiny thing can be addressed after the fact.

Here's a screenshot for reference. I'd expect the menu to stop when the red line makes it to the bottom of the screen when scrolling down but there's still half the old menu distance to go:
![scroll_area](https://github.com/user-attachments/assets/56a0bd5f-e38e-49bf-99ff-9cdb7ceb08e2)

HubertReX commented:
> @HubertReX, looks like this is working as intended, thank you!
> 
> I did notice one tiny thing that came up after the implementation of the keybindings hiding in non-debug modes. That is, when the menu items are hidden in the keybinding list the scroll area isn't being recalculated based on the new number of objects so there's a bunch of blank space underneath.
> 
> Since this isn't affecting functionality in any way I'm going to approve and merge this monster as is and this tiny thing can be addressed after the fact.
> 
> Here's a screenshot for reference. I'd expect the menu to stop when the red line makes it to the bottom of the screen when scrolling down but there's still half the old menu distance to go: 

@DangerousVanilla Yes, I can see it now. I've stumbled upon the full area rect, but haven't noticed its impact on visuals so I left it unchanged. I'll fix it in one small and easy-to-swallow PR ;-)
